eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -165,14 +177,32 @@ actionWellFormed ga = failureUnless isWellFormed $ MalformedProposal ga
       ParameterChange _ ppd -> ppuWellFormed ppd
       _ -> True
 
-govTransition :: forall era. ConwayEraPParams era => TransitionRule (ConwayGOV era)
+govTransition ::
+  forall era.
+  ConwayEraPParams era =>
+  TransitionRule (ConwayGOV era)
 govTransition = do
   -- TODO Check the signatures
   TRC (GovEnv txid epoch, st, gp) <- judgmentContext
 
   let applyProps st' Empty = pure st'
       applyProps st' ((idx, ProposalProcedure {..}) :<| ps) = do
         runTest $ actionWellFormed pProcGovAction
+        -- Check Network for RewardAcnts in ProposalProcedure and its TreasuryWithdrawals
+        expectedNetworkId <- liftSTS $ asks networkId
+        let pProcNetworkId = getRwdNetwork pProcReturnAddr
+        pProcNetworkId
+          == expectedNetworkId
+            ?! ProposalProcedureNetworkIdMismatch pProcNetworkId expectedNetworkId
+        case pProcGovAction of
+          TreasuryWithdrawals wdrls ->
+            forM_ (Map.keys wdrls) $ \rwrdAcnt ->
+              let wdrlNetworkId = getRwdNetwork rwrdAcnt
+               in wdrlNetworkId
+                    == expectedNetworkId
+                      ?! TreasuryWithdrawalsNetworkIdMismatch wdrlNetworkId expectedNetworkId
```

Comment:
We don't need to report a failure for every account, it is better to report a set of accounts that violates this predicate. Also it is better to keep predicate check lazy. This way there is no cost payed upon replay of chain without validation.
```suggestion
            let mismatchedNetworkIds = 
                  Set.filter ((/= expectedNetworkId) . getRwdNetwork) $ Map.keys wdrls
               in Set.null mismatchedNetworkIds
                      ?! TreasuryWithdrawalsNetworkIdMismatch mismatchedNetworkIds expectedNetworkId
```

Reply:
:100: Got it!

Reply:
I didn't notice the suggestion you made here and ended up unifying the `ProposalProcedure` and `TreasuryWithdrawals` failures together. My bad. I will revert that change and apply this suggestion, before re-requesting your feedback :disappointed:

Reply:
In my high-contrast GitHub UI both the suggestions and context-snippets for comments are the same color. I should have payed more attention.

Reply:
I also misread your comment to mean that we need to report `Credential`s for which the `Network` is wrong.

Reply:
:100: Now I got it.